snuba/datasets/events.py

```diff
@@ -97,7 +98,7 @@ class EventsDataset(TimeSeriesDataset):
     and the particular quirks of storing and querying them.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, clickhouse_connection_config: ClickhouseConnectionConfig) -> None:
```

Comment:
What's the reasoning for passing the config during instantiation instead of letting the dataset code pick up the right config from settings? This seems confusing because:
1) whoever needs to get a dataset instance would have to pass a clickhouse config, even if they only need to deal with the datamodel and not run any query.
2) the config passed in input is not really honored since the dataset instance is cached at the first access and not built again (that's why we ensure get_dataset does not take input parameters other than the dataset name)

Reply:
It was an easy way to facilitate overriding the values from the cli commands. That'll probably still need to end up an option here, but yes it can be none most of the time and pickup default values from settigns.